### DISCUSSION TITLE
Mympd no web interface.

### DISCUSSION THREAD

**Discussion by Francy2007** [jcorporation/myMPD#723] [Q&A] [answered]
Hello,
i installed Mympd 9.2.3-1 amd64 and Mpd 0.21.20 on the same pc, with Ubuntu 20.04, but I don't have web interface that connects with localhost: 80.
The configuration in /etc/mympd.conf is as follows:

See the documentation at https://jcorporation.github.io/myMPD/configuration
config mympd main
# WARNING: there can be a lot of writes in the workdir directory
# and if it's on raw flash, that can kill it very fast.
'/ var / lib / mympd' option working directory
mpd_host 'localhost' option
mpd_port option '6600'
acl option ''
http_host 'localhost' option
http_port '80' option
log level of option '5'

    # Requires MYMPD_LUA enabled at compile time
    lualib 'all' option
    scriptacl option '+127.0.0.1'

    # Requires MYMPD_SSL enabled at compile time
    ssl option 'true'
    option port_ssl '443'
    ssl_san option ''
If you do the command "sudo systemctl status mympd" I have the following answer:

● mympd.service - myMPD server daemon
Uploaded: uploaded (/lib/systemd/system/mympd.service; enabled; vendor default: enabled)
Active: Failed (Result: Exit Code) from Mar 2022-04-12 06:07:55 CEST; 21 minutes ago
Process: 1094 ExecStart = / usr / bin / mympd (code = exited, status = 1 / FAILURE)
Main PID: 1094 (code = exit, status = 1 / FAILURE)

Apr 12 06:07:54 vincenzo-imedia-S3800 systemd [1]: Started the myMPD server daemon.
Apr 12 06:07:54 vincenzo-imedia-S3800 mympd [1094]: WARNING mympd Dir work: "/ var / lib / mympd"
Apr 12 06:07:54 vincenzo-imedia-S3800 mympd [1094]: WARNING mympd Config dir: "/ var / lib / mympd / config"
Apr 12 06:07:54 vincenzo-imedia-S3800 mympd [1094]: WARNING mympd Cache dir: "/ var / cache / mympd"
Apr 12 06:07:55 vincenzo-imedia-S3800 mympd [1094]: NOTICE mympd Set loglevel to NOTICE
Apr 12 06:07:55 vincenzo-imedia-S3800 mympd [1094]: WARNING mympd Starting myMPD 9.2.3
Apr 12 06:07:55 vincenzo-imedia-S3800 mympd [1094]: EMERG mympd Unable to connect to http://0.0.0.0:80
Apr 12 06:07:55 vincenzo-imedia-S3800 systemd [1]: mympd.service: Main process terminated, code = exited, status = 1 / FAILURE
Apr 12 06:07:55 vincenzo-imedia-S3800 systemd [1]: mympd.service: failed with result 'exit-code'.
~

How could I solve?

Vincenzo, Italy

**Comment by jcorporation** [comment]
/etc/mympd.conf is not used anymore, read the docs at https://jcorporation.github.io/myMPD/configuration/

The logs seems to be not the original one. How do you installed myMPD? From where you grabbed the package or sources?
And from where is this big warning for the workdir?

Anyway... the log shows that myMPD can not listen on port 80. This is most likely because an other service listens already on this port.

**Reply by Francy2007** [reply]
Hello,
I uninstalled both Mympd, and deleted all related files, and Mpd. I then reinstalled Mympd 9.2.3-1 amd64 from (deb package):
https://download.opensuse.org/repositories/home:/jcorporation/xUbuntu_20.04/amd64/
and Mpd focal, 0.21.20-1build1 amd64 from (deb package):
https://ubuntu.pkgs.org/20.04/ubuntu-universe-amd64/mpd_0.21.20-1build1_amd64.deb.html



I gave the command;
vincenzo @ vincenzo-imedia-S3800: ~ $ sudo mympd -c
[sudo] vincenzo's password:
12:32:37 NOTICE mympd Work dir: "/ var / lib / mympd"
12:32:37 NOTICE mympd Config dir: "/ var / lib / mympd / config"
12:32:37 NOTICE mympd Cache dir: "/ var / cache / mympd"
Created myMPD config and exit
Exiting gracefully, thank you for using myMPD



With the command "systemctl status mympd" I keep getting:

● mympd.service - myMPD server daemon
Loaded: loaded (/lib/systemd/system/mympd.service; enabled; vendor preset: enabled)
Active: failed (Result: exit-code) since Wed 2022-04-13 11:52:50 CEST; 17min Aug.
Process: 1140 ExecStart = / usr / bin / mympd (code = exited, status = 1 / FAILURE)
Main PID: 1140 (code = exited, status = 1 / FAILURE)

Apr 13 11:52:49 vincenzo-imedia-S3800 systemd [1]: Started myMPD server daemon.
Apr 13 11:52:50 vincenzo-imedia-S3800 mympd [1140]: NOTICE mympd Work dir: "/ var / lib / mympd"
Apr 13 11:52:50 vincenzo-imedia-S3800 mympd [1140]: NOTICE mympd Config dir: "/ var / lib / mympd / config"
Apr 13 11:52:50 vincenzo-imedia-S3800 mympd [1140]: NOTICE mympd Cache dir: "/ var / cache / mympd"
Apr 13 11:52:50 vincenzo-imedia-S3800 mympd [1140]: NOTICE mympd Setting loglevel to NOTICE
Apr 13 11:52:50 vincenzo-imedia-S3800 mympd [1140]: NOTICE mympd Starting myMPD 9.2.3
Apr 13 11:52:50 vincenzo-imedia-S3800 mympd [1140]: EMERG mympd Can't bind to http://0.0.0.0:80
Apr 13 11:52:50 vincenzo-imedia-S3800 systemd [1]: mympd.service: Main process exited, code = exited, status = 1 / FAILURE
Apr 13 11:52:50 vincenzo-imedia-S3800 systemd [1]: mympd.service: Failed with result 'exit-code'.

bind_to_addressbind_to_address
I would like to configure the Mympd host, which on Mpd is 192,168,1.231 (bind to address), on / var / lib / mympd / config but with "sudo nano / var / lib / mympd / config" I find it empty. With which command can I configure (with your guide I can not)?

I point out that Mpd with Sonata works.
Best regards.


Vincenzo

**Comment by Francy2007** [comment]
I apologize, I send other questions:

vincenzo@vincenzo-imedia-S3800:~$ sudo nmap -sT -O localhost
[sudo] password di vincenzo: 
Starting Nmap 7.80 ( https://nmap.org ) at 2022-04-13 12:55 CEST
Nmap scan report for localhost (127.0.0.1)
Host is up (0.000060s latency).
Not shown: 995 closed ports
PORT     STATE SERVICE
22/tcp   open  ssh
80/tcp   open  http
631/tcp  open  ipp
9000/tcp open  cslistener
9090/tcp open  zeus-admin
Device type: general purpose
Running: Linux 2.6.X
OS CPE: cpe:/o:linux:linux_kernel:2.6.32
OS details: Linux 2.6.32
Network Distance: 0 hops

OS detection performed. Please report any incorrect results at https://nmap.org/submit/ .
Nmap done: 1 IP address (1 host up) scanned in 2.04 seconds


vincenzo@vincenzo-imedia-S3800:~$ sudo netstat -ntlp | grep LISTEN
tcp        0      0 0.0.0.0:22                 0.0.0.0:*               LISTEN      852/sshd: /usr/sbin 
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      516/cupsd           
tcp        0      0 0.0.0.0:3483             0.0.0.0:*               LISTEN      911/perl            
tcp        0      0 0.0.0.0:9090             0.0.0.0:*               LISTEN      911/perl            
tcp        0      0 0.0.0.0:9000             0.0.0.0:*               LISTEN      911/perl            
tcp        0      0 0.0.0.0:80                 0.0.0.0:*               LISTEN      964/lighttpd        
tcp        0      0 127.0.0.1:5939          0.0.0.0:*              LISTEN      1093/teamviewerd    
tcp        0      0 127.0.0.53:53           0.0.0.0:*              LISTEN      409/systemd-resolve 
tcp6       0      0 :::22                   :::*                             LISTEN      852/sshd: /usr/sbin 
tcp6       0      0 ::1:631                 :::*                           LISTEN      516/cupsd           
tcp6       0      0 :::6600                 :::*                           LISTEN      1/init              
vincenzo@vincenzo-imedia-S3800:~$

**Comment by jcorporation** [comment, marked as answer]
Netstat shows that lighthttp listens already on port 80. Only one service can listen on one port. Disable lighthttp or configure myMPD to listen on an other port. 

/var/lib/mympd/config is a directory, not a file. Inside this directory there are one file for each configuration option. This is described in the documentation.

**Reply by Francy2007** [reply]
Hello, thank you very much.
I uninstalled "lighttpd" and managed to get the Mympd web interface by connecting with 0.0.0.0:80.
But now, since the volume is fixed, I need to know how to control it via Mympd. Should I change the configuration of Mympd or Mpd?

However if I give the command "nano / var / lib / mympd / config" the directory is empty. Should I give a command other than nano?

Vincenzo

**Reply by jcorporation** [reply]
You should edit files inside the directory, not the directory itself, that could not work.

If there is no volume control in myMPD, there is no mixer for the mpd output, that’s a mpd configuration.

**Reply by Francy2007** [reply]
Ok, thank you very much for your availability and help.
Best regards.

Vincenzo
